File: CodingPractice-master/leetcode/algorithms/python/0342_PowerOfFour.py
# -*- coding: utf-8 -*-
# @Author: Yan_Daojiang
# @Date:   2019-05-17 10:39:04
# @Last Modified by:   Yan_Daojiang
# @Last Modified time: 2019-05-17 10:39:13
class Solution:
    def isPowerOfFour(self, num: int) -> bool:
        if num==0:
            return False;
        else:
            while num!=0:
                if num%4!=0 and num!=1:
                    return False
                if num==1:
                    return True
                num=int(num/4)
        return True

File: CodingPractice-master/leetcode/algorithms/cpp/0342_PowerOfFour.cpp
/*
* @Author: Yan_Daojiang
* @Date:   2019-05-17 10:41:45
* @Last Modified by:   Yan_Daojiang
* @Last Modified time: 2019-05-17 10:42:00
*/
class Solution {
public:
    bool isPowerOfFour(int num) {
        if(num==0)
            return false;
        while(num!=0){
            if(num%4!=0&&num!=1)
                return false;
            if(num==1)
                return true;
            num/=4;
        }
        return true;
    }
};